src/core/Expander/Expander/Expander.test.tsx

```diff
@@ -0,0 +1,171 @@
+
+    // re-render the same component with different props
+    rerender(
+      TestExpanderWithProps({
+        children: 'Test expander two',
+      }),
+    );
+    expect(getByTestId('expander-title').textContent).toBe('Test expander two');
+  });
+
+  it('shoud match snapshot', () => {
+    const expanderRenderer = render(TestExpander);
+    const { container } = expanderRenderer;
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it('should not have basic accessibility issues', axeTest(TestExpander));
+});
+
+describe('Custom id', () => {
+  const ExpanderWithCustomId = (
+    <Expander id="test-id" {...{ 'data-testid': 'expander-custom-id' }}>
+      <ExpanderTitle
+        toggleButtonProps={{ 'data-testid': 'expander-custom-id-title' }}
+      >
+        Test expander with custom id
+      </ExpanderTitle>
+      <ExpanderContent {...{ 'data-testid': 'expander-custom-id-content' }}>
+        Test expander with custom id content
+      </ExpanderContent>
+    </Expander>
+  );
+
+  it('is passed on to expander', () => {
+    const { getByTestId } = render(ExpanderWithCustomId);
+    const div = getByTestId('expander-custom-id');
+    expect(div).toHaveAttribute('id', 'test-id');
+  });
+
+  it('is passed on to title', () => {
+    const { getByTestId } = render(ExpanderWithCustomId);
+    const div = getByTestId('expander-custom-id-title');
+    expect(div).toHaveAttribute('id', 'test-id_title');
+  });
+
+  it('is passed on to content', () => {
+    const { getByTestId } = render(ExpanderWithCustomId);
+    const div = getByTestId('expander-custom-id-content');
+    expect(div).toHaveAttribute('id', 'test-id_content');
+  });
+});
+
+describe('defaultOpen', () => {
+  const DefaultOpenExpander = (props?: Omit<ExpanderProps, 'children'>) => (
+    <Expander {...props} defaultOpen={true}>
+      <ExpanderTitle {...{ 'data-testid': 'expander-open-by-default-title' }}>
+        Test expander open by default
+      </ExpanderTitle>
+      <ExpanderContent>Test expander open by default content</ExpanderContent>
+    </Expander>
+  );
+
+  it('gives the classname to expander title and icon', () => {
+    const { getByTestId } = render(DefaultOpenExpander());
+    const div = getByTestId('expander-open-by-default-title');
+    expect(div).toHaveClass('fi-expander_title--open');
+    expect(div.querySelector('svg')).toHaveClass(
+      'fi-expander_title-icon--open',
+    );
+  });
+
+  it('classnames will be removed when clicked', () => {
+    const mockClickHandler = jest.fn();
+    const { getByTestId, getByRole } = render(
+      DefaultOpenExpander({ onOpenChange: mockClickHandler }),
+    );
+    const buttonToClick = getByRole('button');
+    const titleDiv = getByTestId('expander-open-by-default-title');
+    fireEvent.click(buttonToClick);
+    expect(titleDiv).toHaveClass('fi-expander_title');
+    expect(titleDiv).not.toHaveClass('fi-expander_title--open');
```

Comment:
When using the `not.toHaveClass` it might be good to first check **before** the click to make sure that it existed in the first place using `toHaveClass`.

But in this test though, the existence is checked in the previous test. Maybe this is just note to myself :)

Reply:
These are expected in the previous test which uses the same component and props. If this is not sufficient, maybe we could combine the tests, but I'd rather not add duplicate expects.